docs/tutorials/clojure-guide.md

```diff
@@ -0,0 +1,225 @@
+  (yas-global-mode))
+```
+
+For more detailed info on how to setup `lsp-mode`, check [here](https://emacs-lsp.github.io/lsp-mode/page/installation/).
+
+## Installing the language server
+
+`lsp-mode` is a client-server application with `Emacs` acting as the client. In
+order for it to work, you have to install a separate _language server_ that
+understands the specifics of your target language. 
+
+For Clojure, we will use [clojure-lsp](https://github.com/clojure-lsp/clojure-lsp) which as of 2021/01, is the only [LSP](https://microsoft.github.io/language-server-protocol/implementors/servers/) 
+server for Clojure, it has a lot of useful features that I will try to show in this guide.
+
+### Via lsp-mode
+
+`lsp-mode` ships with automatic installation scripts for some of the language servers and clojure-lsp 
+is one of the supported servers, you can install it using <kbd>M-x</kbd> `lsp-install-server` <kbd>RET</kbd> `clojure-lsp`. 
+After installed, `lsp-mode` will automatically initialize it when opening Clojure files.
+
+### Manually
+
+Check [here](https://github.com/clojure-lsp/clojure-lsp#installation) for more information on how to build it manually.
+
+For a custom path, you can set the path of the server location with:
+
+```elisp
+(setq lsp-clojure-custom-server-command '("bash" "-c" "/path/to/clojure-lsp"))
+```
+
+## LSP features
+
+`lsp-mode` has a lot of features and integrations, this guide will show the most important ones 
+that I think it's important on Clojure development. You can check all the features on the 
+official [lsp-mode documentation](https://emacs-lsp.github.io/lsp-mode/)
+
+For more info on `lsp-mode` features, check [here](https://emacs-lsp.github.io/lsp-mode/page/main-features/).
+
+### Syntax check
+
+It checks for errors and warnings on your code while you write it, it uses [clj-kondo](https://github.com/clj-kondo/clj-kondo), 
+another amazing library that lints Clojure code, in the back-end so it's not required to install `clj-kondo` since `clojure-lsp` will
+use it automatically.
+
+It will also check for public unused functions/variables which is something that, as far as I know, does not exist in 
+any other Clojure library/package:
+
+![syntax-checker](images/clojure-syntax-checker.png "syntax-checker")
+
+### Find definition/references
+
+You can find a function definition with `lsp-find-definition` or find all the references to that 
+symbol, variable or function with `lsp-find-references` or `lsp-ui-peek-find-references` if 
+[lsp-ui](https://emacs-lsp.github.io/lsp-ui/) is available. 
+
+![find-references](images/clojure-find-references.gif "find-references")
+
+<details>
+<summary>Doom emacs users</summary>
+
+You may need to remove the lookup handlers conflict from `cider` and/or `clj-refactor` if you want to use this LSP feature.
+
+```elisp
+(use-package! cider
+  :after clojure-mode
+  :config
+  (set-lookup-handlers! 'cider-mode nil))
```

Comment:
What are those lookup handlers?

Reply:
This a `doom-emacs` specific goto-defintion/find-references/etc abstraction.

Reply:
Exactly: https://github.com/hlissner/doom-emacs/blob/eea4709354bf78571c09a1cf49b3c70182bd14f4/modules/tools/lookup/autoload/lookup.el#L4

Reply:
@ericdallo but shouldn't DOOM emacs users use `after!` instead?

Reply:
Not sure @nbfalcon I use this way and it works for a while, what do you suggest?

Reply:
Usually, `after!` is shorter. However, due to `:after`, you'll need something `use-package` line anyway. Ignore the above.

Reply:
Won't it be better to add some more generic instructions instead? I know Doom is popular, but it's not that popular. :-)

Reply:
@bbatsov I think this kind of conflict only happens to doom because of this handler that its use on backend, for vanilla and other frameworks, users just use the keybindings that bind to LSP or cider specific functions, WDYT?